### PR TITLE
BoolOp as loop instead of recursion

### PR DESCRIPTION
**Pull request by maximiliankaul**

### CODE REVIEW

File: cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt [hunk 1]
```diff
@@ -181,43 +181,43 @@ class ExpressionHandler(frontend: PythonLanguageFrontend) :
...
+                val result = newBinaryOperator(operatorCode = op, rawNode = node)
+                var idx = 0
+                var currentInnerNode: BinaryOperator = result
+
+                // For all iterations until the last one: set the lhs to handle the current idx and
+                // create a new nested [BinaryOperator] for the rhs. Then slide the window by one.
+                while (idx < node.values.size - 2) {
+                    currentInnerNode.lhs = handle(node.values[idx])
+                    currentInnerNode.rhs = newBinaryOperator(operatorCode = op, rawNode = node)
+                    currentInnerNode = currentInnerNode.rhs as BinaryOperator
```
Comment (maximiliankaul) [inline cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt:212]: Any Kotlin magic to get rid of this unchecked cast?

Comment (oxisto) [reply cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt:212]: val binOp = newBinaryOperator(operatorCode = op, rawNode = node)
currentInnerNode.rhs = binOp 
currentInnerNode = binOp

File: cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt [hunk 2]
```diff
@@ -181,43 +181,43 @@ class ExpressionHandler(frontend: PythonLanguageFrontend) :
...
         val op =
             when (node.op) {
                 is Python.AST.And -> "and"
                 is Python.AST.Or -> "or"
             }
-        return handleBoolOpRecursively(node.values, op, node)
-    }
+        return when (node.values.size) {
+            0,
+            1 ->
```
Comment (oxisto) [inline cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt:197]: `<= 1` ?

**Review by oxisto — COMMENTED** [review]
Probably a good idea to avoid recursion here, only minor fixes

File: cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt (hunk 1 above)
Comment (maximiliankaul) [reply cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt]: implemented

File: cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt (hunk 2 above)
Comment (maximiliankaul) [reply cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt]: implemented

File: cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt [hunk 3]
```diff
@@ -181,43 +181,43 @@ class ExpressionHandler(frontend: PythonLanguageFrontend) :
...
+                    rawNode = node
+                )
+            else -> {
+                val result = newBinaryOperator(operatorCode = op, rawNode = node)
+                var idx = 0
+                var currentInnerNode: BinaryOperator = result
+
+                // For all iterations until the last one: set the lhs to handle the current idx and
+                // create a new nested [BinaryOperator] for the rhs. Then slide the window by one.
+                while (idx < node.values.size - 2) {
```
Comment (oxisto) [inline cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt:208]: More readable would actually be a for loop until:
`for(idx in 0 until node.values.size - 2)`

Comment (maximiliankaul) [reply cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt:208]: But I also need the `idx` after the loop. Will change to this, if you prefer this style.

Comment (oxisto) [reply cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt:208]: aaah, got it. in this case you cant use for

**Comment by oxisto** [comment]
We will take the solution in #1698

**Pull request closed without merging**